src/ngraph/op/interpolate.hpp

```diff
@@ -22,18 +22,18 @@ namespace ngraph
 {
     namespace op
     {
-        typedef struct
-        {
-            AxisSet axes;
-            std::string mode;
-            bool align_corners = true;
-            bool antialias = false;
-            std::vector<size_t> pads_begin;
-            std::vector<size_t> pads_end;
-        } InterpolateAttrs;
-
         namespace v0
         {
+            typedef struct
+            {
+                AxisSet axes;
+                std::string mode;
```

Comment:
Can we use an enum here? Using string is very unsafe.
This struct could use some documentation since it is a parameter to a function.

Reply:
I agree enum is better but I followed how v0 done.

Reply:
Added an Enum. The string will be check against the Enum in the constructor.

Reply:
v0 already has users based on the this struct. so could not change it.

Reply:
Also added unit-test to check for invalid string.

Reply:
Changed to enums for v3. v0 is still strings.

Reply:
Okay. Best not to mess with the previous version of the op.